apisix/plugins/redirect.lua

```diff
@@ -147,6 +148,7 @@ function _M.rewrite(conf, ctx)
     core.log.info("plugin rewrite phase, conf: ", core.json.delay_encode(conf))
 
     local ret_code = conf.ret_code
+    local ret_port = tonumber(ctx.var["var_x_forwarded_port"])
```

Comment:
Recently, I upgraded apisix with the latest code, and found that there was an error in converting http to https. When redirecting, I always bring a port 80. It should be caused by this change, because I am listening on port 80 and port 443, but ngx_tpl There is a piece of code in it that is "set $var_x_forwarded_port $server_port;", which means that the port from http to https is always the same. I don't think this change is very reasonable?

Reply:
Oh, seems I got what you mean. The server port `80` and then enable `http_to_https`, it'll always redirect to `https://host:80` then get `400 bad request`. cc @tokers @spacewander @tzssangglass 

We need to take more consideration about https://github.com/apache/apisix/pull/6686#issuecomment-1113004299

Reply:
Can we set a default value for this case?
If there is no `X-Forwarded-Port` in clinet reuqest headers, then the `redirect` plugin use `443` as the default https port?

Reply:
It has default as the server_port

https://github.com/apache/apisix/blob/1b5c1900da517bb7d34d4b221aace7242345e78a/apisix/cli/ngx_tpl.lua#L668

As mentioned before https://github.com/apache/apisix/pull/6686#issuecomment-1112812875

Reply:
> It has default as the server_port

I mean in the `redirect` plugin, if someone enable `http_to_https` and the client request doesn't pass X`-Forwarded-Port`, use 443 as the default port in `Location` and don't care about the `server_port` variable.

Reply:
As the conversation before https://github.com/apache/apisix/pull/6686#issuecomment-1075798013

The original implement with an extra attribute `ret_port`, but no acceptable.

Reply:
Maybe we can read the correct port from `config.yaml`?

Reply:
把这个set $var_x_forwarded_port       $server_port;   改成set $var_x_forwarded_port       443;